packages/runtime/src/widgets/PopupMenu.tsx

```diff
@@ -0,0 +1,107 @@
+import type { ReactNode } from "react";
+import React, { isValidElement, useEffect, useMemo, useState } from "react";
+import { Menu, Dropdown as AntdDropdown, Button } from "antd";
+import MoreVertOutlinedIcon from "@mui/icons-material/MoreVertOutlined";
+import { cloneDeep, merge } from "lodash-es";
+import {
+  type EnsembleWidget,
+  useExecuteCode,
+  useRegisterBindings,
+  unwrapWidget,
+} from "framework";
+import { useNavigateScreen } from "../runtime/navigate";
+import type { EnsembleWidgetProps } from "../util/types";
+import { WidgetRegistry } from "../registry";
+
+interface PopupMenuItem {
+  label: string;
+  value: string;
+}
+
+interface PopupMenuStyles {
+  backgroundColor: string;
+  borderRadius: string;
+  borderColor: string;
+  borderWidth: string;
+}
+type PopupMenuProps = {
+  [key: string]: unknown;
+  items: PopupMenuItem[];
+  widget?: Record<string, unknown>;
+  styles?: PopupMenuStyles;
+  onItemSelect?: {
+    executeCode?: string;
+    navigateScreen?: string;
+  };
+} & EnsembleWidgetProps;
+
+const defaultStyles: PopupMenuStyles = {
+  backgroundColor: "transparent",
+  borderRadius: "50%",
+  borderWidth: "1px",
+  borderColor: "transparent",
+};
+export const PopupMenu: React.FC<PopupMenuProps> = (props) => {
+  const onItemSelect = props.onItemSelect?.executeCode;
+  const mergedStyles = merge(defaultStyles, props.styles);
+  const { values } = useRegisterBindings(props, props.id);
+  const onTapCallback = useExecuteCode(onItemSelect, values);
+  const onNavigate = useNavigateScreen(props.onItemSelect?.navigateScreen);
+  const [widgetProps, setWidgetProps] = useState<EnsembleWidget>();
+  const menuItems = (
+    <Menu>
+      {props.items.map((item) => (
+        <Menu.Item
+          key={item.value}
+          onClick={onItemSelect ? onTapCallback : onNavigate}
```

Comment:
You will need to pass in the value to the callback so you know which item was selected

Reply:
I'll have an update to the actions hook that will make this easier.